### PR TITLE
Ignore sounds while safe mode is on

### PR DESCRIPTION
**Pull request by Ramza13**
#### Summary
SUMMARY: Interface "Ignore sounds while safe mode is on"

#### Purpose of change
Part of #28672
While safe mode is active you will not be prompted to stop what you are doing due to sounds.

#### Describe the solution
Used #28961 but with safe mode instead instead of an option

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
Ideally we need some sort of an upgraded safe mode manager in which user would be able to manually set distraction types he wants to ignore.
But for now even only one ignorable distraction type is better than current status quo.

File: src/sounds.cpp [hunk 1]
```diff
@@ -427,7 +427,8 @@ void sounds::process_sound_markers( player *p )
         }
 
         // don't print our own noise or things without descriptions
-        if( !sound.ambient && ( pos != p->pos() ) && !g->m.pl_sees( pos, distance_to_sound ) ) {
+         if( !sound.ambient && ( pos != p->pos() ) && !g->m.pl_sees( pos, distance_to_sound ) &&
```
Comment (kevingranade) [inline src/sounds.cpp]:
```suggestion
        if( !sound.ambient && ( pos != p->pos() ) && !g->m.pl_sees( pos, distance_to_sound ) &&
```

**Comment by Amoebka** [comment]
Isn't this really confusing? Turning safe mode on REMOVES warnings for potential danger.

**Comment by Ramza13** [comment]
> Isn't this really confusing? Turning safe mode on REMOVES warnings for potential danger.

So one possibility I was considering was adding the ability to make sound based safe mode rules so that you could have safe mode trigger on certain sounds to give you more control of this behavior.  That way you could determine how exactly you wanted to be warned about sound.  Do you think that would help?

**Comment by Destragon** [comment]
I agree with Amoebka, the interruption prompts should come up during safe mode, not the other way around.

**Comment by Ramza13** [comment]
> I agree with Amoebka, the interruption prompts should come up during safe mode, not the other way around.

I'm not sure that really works either.  For one thing safe mode turns back on automatically if you use that option which would get annoying.  For the other, I want to make sure the player is safe while doing these things and safe mode is part of that.

That said I am still planning on trying some version of this, I close this because I wasn't satisfied with this solution, I am going to keep trying.

**Pull request closed without merging**